src/eval.rs

```diff
@@ -774,6 +774,16 @@ fn reduce_with_witness<F: LurkField>(
                     } else {
                         Control::Return(arg1, env, store.intern_cont_unop(Op1::Char, cont))
                     }
+                } else if head == store.sym("eval") {
+                    let (arg1, end) = match store.car_cdr_mut(&rest) {
+                        Ok((car, cdr)) => (car, cdr),
+                        Err(e) => return Err(LurkError::Reduce(e.into())),
+                    };
+                    if !end.is_nil() {
+                        Control::Return(expr, env, store.intern_cont_error())
+                    } else {
+                        Control::Return(arg1, env, store.intern_cont_unop(Op1::Eval, cont))
+                    }
```

Comment:
The lisper in me wants to write 

```rust
Control::Return(expr, env, {if !end.is_nil() { store.intern_cont_error() else store.intern_cont_unop(Op1::Eval, cont)}})
```
but i'm not sure if that's idiomatic rust.